src/main/java/Duke.java

```diff
@@ -1,10 +1,245 @@
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.time.LocalTime;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.File;
+
 public class Duke {
     public static void main(String[] args) {
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("What can I do for you?\n");
+
+        boolean isOnline = true;
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        File savedTasks = new File("saved_tasks_list.txt");
+        try {
+            if (!savedTasks.createNewFile()) {
+                // Load the exist file
+                Scanner fileScanner = new Scanner(savedTasks);
+                while (fileScanner.hasNextLine()) {
+                    String record = fileScanner.nextLine();
+                    String[] fields = record.split("\\|");
+                    Task currTask;
+                    if (fields[0].equals("T")) {
+                        currTask = new Todo(fields[2]);
+                    } else if (fields[0].equals("D")) {
+                        currTask = new Deadline(fields[2], fields[3]);
+                    } else if (fields[0].equals("E")) {
+                        currTask = new Event(fields[2], fields[3], fields[4]);
+                    } else {
+                        currTask = new Task("");
+                    }
+
+                    if (fields[1].equals("1")) {
+                        currTask.setDone();
+                    } else {
+                        currTask.setNotDone();
+                    }
+                    taskList.add(currTask);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        while (isOnline) {
+            try {
+                isOnline = takeRequest(taskList);
+            }
+            catch (DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+
+        try {
+            FileWriter writer = new FileWriter("saved_tasks_list.txt");
+            for (int i = 0; i < taskList.size(); i++) {
+                writer.write(taskList.get(i).toRecord() + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
-}
+
+    public static boolean takeRequest(ArrayList<Task> taskList) throws DukeException {
+        Scanner keyboard = new Scanner(System.in);
+        String userInput;
+
+        while (true) {
+            userInput = keyboard.next();
+            System.out.println("--------------------------------");
+            switch (userInput) {
```

Comment:
I think the Java standard we follow doesn't have indentations for each `case`. No need for Egyptian braces within the cases too! Something like:
```java
switch (userInput) {
case "bye":
    ...
case "list":
    ...
...
```